app/components/elements/Button.tsx

```diff
@@ -0,0 +1,31 @@
+interface ButtonProps {
+  className?: string;
+  children?: React.ReactNode;
+  name?: string;
+  value?: string;
+  type?: 'button' | 'submit' | 'reset';
+}
```

Comment:
Let's have these `ButtonProps` extend the actual props for a `<button>` element. An example of how to do this with TypeScript is here: https://stackoverflow.com/a/62568833/715870

Then, you can delete the members of this `ButtonProps` interface that simply repeat existing props — in this case, that appears to be every member. However, what we can then do to make this really useful is add something like:

```tsx
interface ButtonProps {
  size?: 'small' | 'medium' | 'large' // We'll want to set 'medium' as the default later if this isn't passed
  appearance?: 'primary' | 'secondary' | 'danger' | 'success' // Set different colours based on this prop
}
```

Hopefully this makes sense. Let me know if not!